label/service.go

```diff
@@ -0,0 +1,282 @@
+package label
+
+import (
+	"context"
+	"strings"
+
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/kv"
+)
+
+type Service struct {
+	store *Store
+	kvSvc *kv.Service
+}
+
+func NewService(st *Store, kvSvc *kv.Service) influxdb.LabelService {
+	return &Service{
+		store: st,
+		kvSvc: kvSvc, // todo (al) this can be removed once URMs are removed from the Label service
+	}
+}
+
+// CreateLabel creates a new label.
+func (s *Service) CreateLabel(ctx context.Context, l *influxdb.Label) error {
+	if err := l.Validate(); err != nil {
+		return &influxdb.Error{
+			Code: influxdb.EInvalid,
+			Err:  err,
+		}
+	}
+
+	l.Name = strings.TrimSpace(l.Name)
+
+	err := s.store.Update(ctx, func(tx kv.Tx) error {
+		if err := uniqueLabelName(ctx, tx, l); err != nil {
+			return err
+		}
+
+		if err := s.store.CreateLabel(ctx, tx, l); err != nil {
+			return err
+		}
+		if err := s.kvSvc.CreateUserResourceMappingForOrg(ctx, tx, l.OrgID, l.ID, influxdb.LabelsResourceType); err != nil {
```

Comment:
```suggestion
	urmCreator UserResourceMappingCreator
}

type UserResourceMappingCreator interface {
    CreateUserResourceMappingForOrg(ctx context.Context, orgID, id, *influxdb.ID, influxdb.ResourceType) error
}

func NewService(st *Store, urmCreator UserResourceMappingCreator) influxdb.LabelService {
	return &Service{
		store: st,
		urmCreator: urmCreator,
	}
}

// CreateLabel creates a new label.
func (s *Service) CreateLabel(ctx context.Context, l *influxdb.Label) error {
	if err := l.Validate(); err != nil {
		return &influxdb.Error{
			Code: influxdb.EInvalid,
			Err:  err,
		}
	}

	l.Name = strings.TrimSpace(l.Name)

	err := s.store.Update(ctx, func(tx kv.Tx) error {
		if err := uniqueLabelName(ctx, tx, l); err != nil {
			return err
		}

		if err := s.store.CreateLabel(ctx, tx, l); err != nil {
			return err
		}
		if err := s.urmCreator.CreateUserResourceMappingForOrg(ctx, tx, l.OrgID, l.ID, influxdb.LabelsResourceType); err != nil {
```

Reply:
@GeorgeMac is this what your suggestiong?

Reply:
Almost. I am suggesting `CreateUserResourceMappingForOrg` does not take a `kv.Tx` and instead it is defined back in `kv` package and starts an `Update` and call the unexported one with a `kv.Tx`.
Then here we could call `CreateUserResourceMappingForOrg(ctx, l.OrgID, l.ID, influxdb.LabelsResourceType)` without the tx and we would do so outside of this current transaction.

Reply:
Moving the last line outside this current `Update` transaction, so we can go through the URM service front door. Ultimately, I we will remove this anyway in the next couple weeks as well.

Reply:
I added a suggestion below for how to change `kv` to support this.

Reply:
Okay so I just implemented this as well as the wrapper that Lyon suggested!